### DISCUSSION TITLE
Help getting bmc4 working - New to docker

### DISCUSSION THREAD

**Discussion by Apok1** [itzg/docker-minecraft-server#2821] [Q&A]
Hi all,

Thank you for th efforts in putting this all together and maintaining everything.

1/2 - Noob question, but am I correct in thinking you run the compose once with SETUP_ONLY as true and then again with false?  I assume so as I got the example yaml from [here, bmc4](https://github.com/itzg/docker-minecraft-server/blob/master/examples/bmc4/docker-compose.yml) working, though I could not connect with the client as it's default is V4 and my client is v26 (and forge, but apparently that should be fine). I hope to move my v26 world in once it's working.

If I change MODRINTH_VERSION to v26 however it fails to start the server, ending with "Done".  I have chnaged as little as possible and my compose looks like this:

services:
  mc:
    image: itzg/minecraft-server
    environment:
      EULA: true
      TYPE: MODRINTH
      MEMORY: 4G
      MODRINTH_MODPACK: better-mc-forge-bmc4
      MODRINTH_VERSION: v26 # changed to 26
      MODRINTH_FORCE_SYNCHRONIZE: true
      MODRINTH_EXCLUDE_FILES: |
        citresewn
        oculus
        yungsmenutweaks
        notenoughanimations
        ryoamiclights
        euphoriapatcher
        entity_model_features
        entity_texture_features
        3dskinlayers
      MODRINTH_OVERRIDES_EXCLUSIONS: |
        mods/NekosEnchantedBooks-*.jar
        mods/citresewn-*.jar
      DEBUG: true
      SETUP_ONLY: true
    volumes:
      - /mnt/cache/mineserv:/data  # Changed to my path
    ports:
      - "25565:25565"

Am I missing something, if not maybe the excluded files/overrides need updating, but not sure how would I go about doing this?  if nothing else does anyone have a working compose for v26 based on MODRINTH or autoforge?

Appreciate any help you can offer.

**Comment by itzg** [comment]
Sorry, for the confusion with SETUP_ONLY. I had accidentally left that there from testing. Normally you just don't include that variable. (That option is for non-docker users that just want to leverage the setup logic)

As for the issue with v26 I would need to see some logs. In general though, this is something that the modpack publisher should really help with since they are not designating mod client/server applicability correctly.

**Reply by itzg** [reply]
Actually, I'm guessing the image now needs to be `itzg/minecraft-server:java17` since the latest java version, 21, is probably not compatible with Forge 1.20.1

**Reply by itzg** [reply]
Example file has been updated and now works

https://github.com/itzg/docker-minecraft-server/blob/master/examples/bmc4/docker-compose.yml

For some reason the `connector` is trying to load Fabric things within Forge. I added it to the exclusions. 

I also cleaned up those left over testing variables.

**Comment by Apok1** [comment]
Hi itzg, amazing!! - Thank you so much, absolute legend for this and the entire project.

**Comment by MisteRice67** [comment]
Hello,

I'm an absolute beginner when it comes to NAS and Docker. I've managed to download the docker-compose.yml file from the docker-minecraft-server/examples/bmc4 directory and set up the server. The server appears to start up fine, and everything seems to be in order so far.

owever, when I try to connect to the server using the CurseForge Launcher v26, my screen freeze during the encrypting. 
What can I do to fix this? What should I change?

services:
  mc:
    image: itzg/minecraft-server:java17
    environment:
      EULA: true
      TYPE: MODRINTH
      MEMORY: 18G
      MODRINTH_MODPACK: better-mc-forge-bmc4
      MODRINTH_VERSION: v26
      MODRINTH_EXCLUDE_FILES: |
        citresewn
        oculus
        yungsmenutweaks
        notenoughanimations
        ryoamiclights
        euphoriapatcher
        entity_model_features
        entity_texture_features
        3dskinlayers
        connector
      MODRINTH_OVERRIDES_EXCLUSIONS: |
        mods/NekosEnchantedBooks-*.jar
        mods/citresewn-*.jar
    volumes:
      - /volume1/docker/MC-Server/.data
    ports:
      - "25565:25565"

volumes:
  data: {}
  
  
#Minecraft server properties
allow-flight=true
allow-nether=true
broadcast-console-to-ops=true
broadcast-rcon-to-ops=true
difficulty=hard
enable-command-block=true
enable-jmx-monitoring=false
enable-query=true
enable-rcon=false
enable-status=true
enforce-secure-profile=true
enforce-whitelist=true
entity-broadcast-range-percentage=100
force-gamemode=false
function-permission-level=2
gamemode=survival
generate-structures=true
generator-settings={}
hardcore=false
hide-online-players=false
initial-disabled-packs=
initial-enabled-packs=vanilla
level-name=world
level-seed=
level-type=default
max-build-height=256
max-chained-neighbor-updates=1000000
max-players=6
max-tick-time=120000
max-world-size=29999984
motd=Better MC [FORGE] 1.20
network-compression-threshold=256
online-mode=true
op-permission-level=4
player-idle-timeout=0
port=25565
prevent-proxy-connections=false
pvp=true
query.port=10081
rate-limit=0
rcon.password=
rcon.port=10085
require-resource-pack=false
resource-pack=
resource-pack-prompt=
resource-pack-sha1=
server-ip=***
server-name=***
server-port=25565
simulation-distance=10
snooper-enabled=false
spawn-animals=true
spawn-monsters=true
spawn-npcs=true
spawn-protection=16
sync-chunk-writes=true
text-filtering-config=
use-native-transport=true
view-distance=10
white-list=true


Appreciate any help.

**Reply by MisteRice67** [reply]
Update:

Using MODRINTH_Launcher and its not stucking in encrypting.
Now the client shut down after showing "Saving World".

[Server thread/INFO] [xa.pa.OpenPartiesAndClaims/]: Updating all force
load tickets for 2fd3f02e-5f82-4fbf-931c-373296191a4d

Is in the logs.

Already removed OpenPartiesAndClaims and 
Xaeros_Minimap
XaerosWorldMap

still not working.

**Reply by MisteRice67** [reply]
okay new update:

I was able to join on the server. After 1 min. running through the world I was kicked and the log:

[Server thread/INFO] [xa.pa.OpenPartiesAndClaims/]: Updating all force
load tickets for 2fd3f02e-5f82-4fbf-931c-373296191a4d 

appears again

**Comment by Apok1** [comment]
Hi, 

Bring that memory down to 4G and try again, that's the easiest thing to try.   More memory is not always better in MC servers.

It could be a client / server modpack issue.  I had issues as I was using an existing world that was made in curse forge.  I switched to curse forge and it's been rock solid since.

Double check and reinstall the BMC4 MODRINTH launcher and modpack, make sure you use the MODRINTH modpack and not the curse forge one for your client.  they are not quite the same.

Good luck!

**Reply by MisteRice67** [reply]
version: '3'

services:
  mc:
    image: itzg/minecraft-server:java17
    container_name: better-minecraft
    environment:
      EULA: "TRUE"
      TYPE: CURSEFORGE
      MEMORY: 18G
      CF_SERVER_MOD: /data/ServerMods.zip
      CF_BASE_DIR: /data
      CF_DOWNLOAD: https://mediafilez.forgecdn.net/files/4607/433/Better+MC+FORGE+1.20.1+v26.zip
    ports:
      - "25565:25565"
    volumes:
      - ./data:/data
    restart: unless-stopped

volumes:
  data:
  
  Yep that fixed that problem :)

**Comment by skrekkugle** [comment]
I'm stuck at encrypting when joining the server. Using the bmc4 docker-compose.yml as is, downloaded BMC4 in Modrinth, and launching with Modrinth. Any ideas?

**Reply by MisteRice67** [reply]
version: '3'

services:
  mc:
    image: itzg/minecraft-server:java17
    container_name: BMC4
    environment:
      EULA: "TRUE"
      TYPE: FABRIC  # Assuming you're using Fabric for Modrinth
      MEMORY: 4G
      MODRINTH_SERVER_MOD: /data/BMC4_FABRIC_1.20.1_Server_Pack.zip  # Adjust this to your Modrinth server pack
      MODRINTH_BASE_DIR: /data
      MODRINTH_DOWNLOAD: https://modrinth.com/modpack/better-mc-fabric-bmc4/download/5399131  # Example URL, replace with actual Modrinth URL
    ports:
      - "25565:25565"
    volumes:
      - ./data:/data
    restart: unless-stopped

volumes:
  data:
  
  
    YPE is set to FABRIC, which is commonly used with Modrinth modpacks.
    MODRINTH_SERVER_MOD should point to the server pack file you have downloaded for Modrinth.
    MODRINTH_BASE_DIR is set to /data to match the volume mount.
    MODRINTH_DOWNLOAD should be updated with the correct download URL for your Modrinth modpack.

Ensure that you have the correct Modrinth server pack and URL. You may need to adjust the TYPE and other settings based on the specific requirements of the Modrinth modpack you are using.



Try these

**Reply by skrekkugle** [reply]
Thanks for the quick reply! Seems like my problem was that I was using V27.5 of the mod instead of V26 as in the docker-compose file. New to Modrinth so wasn't aware of the Versions tab. Thanks anyways!
